java/rocksjni/options.cc

```diff
@@ -232,7 +232,7 @@ void Java_org_rocksdb_Options_setStatistics(
     JNIEnv* env, jobject jobj, jlong jhandle, jlong jstatistics_handle) {
   auto* opt = reinterpret_cast<rocksdb::Options*>(jhandle);
   auto* pSptr =
-      reinterpret_cast<std::shared_ptr<rocksdb::StatisticsJni>*>(
+      reinterpret_cast<std::shared_ptr<rocksdb::Statistics>*>(
```

Comment:
I am just wondering about the purpose of changing `StatisticsJni` to `Statistics` here?

Reply:
Reinterpret_cast is safe to use when a pointer converted to an integer type is converted back to pointer of same type. In the recent iteration, I changed the newStatistics() method to return a pointer to shared_ptr<rocksdb::Statistics> instead of shared_ptr<rocksdb::StatisticsJni> which would ensure safety while dereferencing the object.